receiver/receiver.go

```diff
@@ -23,201 +20,66 @@ var (
 
 // Metrics receiver receives metrics.
 // Its purpose is to translate data from any format to the collector's internal metrics format.
 // MetricsReceiver feeds a consumer.Metrics with data.
 //
 // For example, it could be Prometheus data source which translates Prometheus metrics into pmetric.Metrics.
-type Metrics interface {
-	component.Component
-}
+type Metrics = internal.Metrics
 
 // Logs receiver receives logs.
 // Its purpose is to translate data from any format to the collector's internal logs data format.
 // LogsReceiver feeds a consumer.Logs with data.
 //
 // For example, it could be a receiver that reads syslogs and convert them into plog.Logs.
-type Logs interface {
-	component.Component
-}
+type Logs = internal.Logs
 
 // CreateSettings configures Receiver creators.
 //
 // Deprecated: [v0.103.0] Use receiver.Settings instead.
-type CreateSettings = Settings
+type CreateSettings = internal.Settings
 
 // Settings configures Receiver creators.
-type Settings struct {
-	// ID returns the ID of the component that will be created.
-	ID component.ID
-
-	component.TelemetrySettings
-
-	// BuildInfo can be used by components for informational purposes.
-	BuildInfo component.BuildInfo
-}
+type Settings = internal.Settings
 
 // Factory is factory interface for receivers.
 //
 // This interface cannot be directly implemented. Implementations must
 // use the NewReceiverFactory to implement it.
-type Factory interface {
-	component.Factory
-
-	// CreateTracesReceiver creates a TracesReceiver based on this config.
-	// If the receiver type does not support tracing or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
-	CreateTracesReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Traces) (Traces, error)
-
-	// TracesReceiverStability gets the stability level of the TracesReceiver.
-	TracesReceiverStability() component.StabilityLevel
-
-	// CreateMetricsReceiver creates a MetricsReceiver based on this config.
-	// If the receiver type does not support metrics or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
-	CreateMetricsReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Metrics) (Metrics, error)
-
-	// MetricsReceiverStability gets the stability level of the MetricsReceiver.
-	MetricsReceiverStability() component.StabilityLevel
-
-	// CreateLogsReceiver creates a LogsReceiver based on this config.
-	// If the receiver type does not support the data type or if the config is not valid
-	// an error will be returned instead. `nextConsumer` is never nil.
-	CreateLogsReceiver(ctx context.Context, set Settings, cfg component.Config, nextConsumer consumer.Logs) (Logs, error)
-
-	// LogsReceiverStability gets the stability level of the LogsReceiver.
-	LogsReceiverStability() component.StabilityLevel
-
-	unexportedFactoryFunc()
-}
+type Factory = internal.Internal
```

Comment:
Why not `internal.Factory`?

Reply:
I followed what @mx-psi did in this PR: https://github.com/open-telemetry/opentelemetry-collector/pull/10253/files#diff-56869497b872826bc1bd89731571d212be40aa7fe7252864a80317e6413234e3R126

Since there is a private `factory` struct, I guess it makes sense to use an entirely different name to avoid mixing them.

Reply:
No strong opinion on the name, if `Factory` is preferred I am fine with that

Reply:
I have renamed `internal.Internal` to `internal.Factory`.